src/server/routes/parcel.routes.ts

```diff
@@ -45,26 +85,28 @@ export default function registerRoutes(
       });
 
       const activeEndpoint = await fastify.getActiveEndpoint();
-      try {
-        await activeEndpoint.validateMessage(parcel);
-      } catch (err) {
-        parcelAwareLogger.info({ err }, 'Refusing invalid parcel');
+      const serviceOptions: ServiceOptions = {
+        logger: parcelAwareLogger,
+        dbConnection: fastify.mongoose,
+        activeEndpoint,
+      };
```

Comment:
This is an overkill in my opinion -- each function call should only be passed the stuff it actually needs.